src/v/cloud_storage/remote_segment.cc

```diff
@@ -225,15 +277,15 @@ ss::future<> remote_segment::run_hydrate_bg() {
     }
 }
 
-ss::future<std::filesystem::path> remote_segment::hydrate() {
+ss::future<> remote_segment::hydrate() {
     return ss::with_gate(_gate, [this] {
         auto full_path = _manifest.get_remote_segment_path(_path);
         vlog(_ctxlog.debug, "segment {} hydration requested", full_path);
-        ss::promise<std::filesystem::path> p;
+        ss::promise<ss::file> p;
         auto fut = p.get_future();
         _wait_list.push_back(std::move(p), ss::lowres_clock::time_point::max());
         _bg_cvar.signal();
-        return fut;
+        return fut.then([this](ss::file f) { _data_file = std::move(f); });
```

Comment:
Looks like we set `_data_file` in `run_hydrate_bg` and overwrite it (with the same handle?) here. This probably can simply be a void future?